### PR TITLE
feat: host-based GraphQL context

### PR DESCRIPTION
**Pull request by bchrobot**
This extends the current GraphQL context to include database and memoizer clients. Currently these will always be the same but could be extended in the future for per-host contexts.

We probably won't do the big schema-based multitenancy refactor but we might as well merge the work from this year-old PR before it goes stale.

Needs to be rebased on `master` after #726 is merged.

### CODE REVIEW

**Review by henryk1229 — APPROVED** [review]

File: src/server/contexts.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+import createMemoizer from "memoredis";
+
+import { config } from "../config";
+import logger from "../logger";
+import { createLoaders, r } from "./models";
+import { Memoizer } from "./types";
+
+export interface SpokeDbContext {
+  schema: string;
+  master: Knex;
```
Comment (ben-pr-p) [inline src/server/contexts.ts:13]:
```suggestion
  primary: Knex;
```
The world seems to be going towards this, we should follow?

**Review by ben-pr-p — CHANGES_REQUESTED** [review]
This is incredibly close to full schema tenancy and looks great overall!

**Comment by ben-pr-p** [comment]
Definitely maybe later!

**Review by ben-pr-p — APPROVED** [review]

**Comment by ben-pr-p** [comment]
I think we should do a release after merging this, before merging too much else

**Pull request merged**